gcp/api/server.py

```diff
@@ -340,9 +421,14 @@ def do_query(query, context: grpc.ServicerContext, include_details=True):
     ecosystem = ''
     purl_str = ''
 
-  page_token = None
+  context.page_token = None
   if query.page_token:
-    page_token = ndb.Cursor(urlsafe=query.page_token)
+    try:
+      context.page_token = ndb.Cursor(urlsafe=query.page_token)
```

Comment:
Can we do this earlier as part of constructing the QueryContext instead?

Reply:
Not really, since the page token exists in each query within a batch_query, not just one page token for the entire batch_query

Reply:
Discussed offline: looks like we can just initialise this earlier.